### DISCUSSION TITLE
Experienced shortcomings in routing for walking.

### DISCUSSION THREAD

**Discussion by Kamerijker** [organicmaps/organicmaps#7384] [Ideas]
Hallo,

I tried OM extensively for walking in several countries over the last year, and I like the user interface, and general usability above many of the competitors apps.
I also like the possibility of downloading smaller local maps, that limit the use of memory.
I would  welcome more dense contour lines, even if that takes more memory space.

However I am a little disappointed in some routing shortcomings or options that OM takes, in comparison with other established apps for walking.

1.	OM does not support oneway for pedestrians on paths (preferred tag foot:backward=no). This is important for theme parks or other restricted areas where dedicated entrance and exit can be rather far away from each other, and as such result in routing through the wrong path and getting pedestrians to the wrong spot. It may also apply to paths who are designated oneway for traffic reasons or other, like narrow passings etc. All of the competitors routers I tried did ok here.

2.	OM prohibits pedestrian passing through a road whose key is access=destination, unless specifically tagged as foot=yes. This can result in very strange routings indeed. In hardly any country (none that I know off ) the qualification destination applies to pedestrians. So pedestrians should always have access unless otherwise specified as foot=destination. Although the OSM guidelines prefer not using this general key and tagging more specifically for vehicles or motor vehicles, there are still thousands of instances where the general tag is used and it would be very unrealistic to expect they’ll all be changed. Therefore most competitor apps disregard the general destination restriction for pedestrians and that is ok.

3.	OM tries to avoid primary roads by default when routing for foot. Here again the bulk of primary roads do not restrict the use by pedestrians unless specifically forbidden. Avoiding them by default results sometime in very awkward routings especially in an urban environment, because primary roads (usually national roads) mostly extend to the centre of cities. Therefore primary roads should always be allowed unless specifically tagged as foot=no. Many of the competitors apps provide an option to avoid primary or busy roads and that would be preferred above the default as taken by OM.

I hope that these few remarks can be taken into account in one of the next releases of OM because it is one of the most usable and user friendly products for walking on the market.

I would welcome the experience of some other users in these areas.

**Comment by biodranik** [comment]
Thanks for the feedback! Are there already existing issues for mentioned points? Please create (or update) them with examples from OSM, so we can work separately on each one.

**Reply by Kamerijker** [reply]
Thanks for the comment.
My first point (oneway for footpaths) seems to be discussed in issue #5136 but is not resolved.
I will add some examples.
My second point (access=destination and pedestrian routing) and third point (pedestrian routing in relation to primary roads) are probably new and I will open new issues.
